### ISSUE TITLE
Add a confirmation modal when clearing the customizations on a template

### ISSUE THREAD

**Issue by hanneslsm** [WordPress/gutenberg#39124]
## What problem does this address?
The impact on clearing a customization on a template or template part can be huge, especially when you put a lot work in building your templates with the FSE.
However, all the work can be deleted with just one or two thoughtless clicks on the menu:
<img width="244" alt="image" src="https://user-images.githubusercontent.com/17861443/155993533-9c814d5d-279d-4350-9093-99f05963e52c.png"> 

## What is your proposed solution?
Add a modal where the user has to confirm that all their work will be deleted. Maybe some additional info text to make clear what impact the clearing exactly has.

**Comment by ntsekouras** [comment]
I agree that the impact might be big and it could be good to have an extra confirmation step.. --cc @jameskoster @jasmussen @kjellr

**Comment by jasmussen** [comment]
Keeping #25071 in mind, it seems okay to add a confirmation dialog. I'd keep it simple, though. Maybe use the new ConfirmDialog from #34153?

**Comment by jameskoster** [comment]
Worth noting that when you revert _whilst editing the document_ there is an undo affordance:

https://user-images.githubusercontent.com/846565/156340717-69d8283b-87d0-4d40-bd66-4cdbbafb8994.mp4

We used the undo pattern here for a couple of reasons:

1. It aligns with other flows in WP like trashing posts
2. It essentially allows you to "preview" the revert directly on the canvas before fully committing 

Obviously the second point doesn't apply on the template list view, and the first point holds less water generally since we removed the ability to retrieve templates from the trash.

All that to say: I think it's probably okay to install the confirmation dialog on template list, but we should acknowledge that the revert flow would then differ based on the view. Is it risky to make a high impact action feel unpredictable?

If we wanted to make reverting consistent, another option might be to make the button itself feel more like a "danger zone" (IE make the link red, to match Trash links), and add the undo action to the Snackbar on list view.

**Comment by afercia** [comment]
Worth mentioning that also when _deleting_ a template (e.g. a custom template) there's no confirmation step. Is there a specific issue for that already?

<img width="1117" alt="Screenshot 2022-07-19 at 13 43 50" src="https://user-images.githubusercontent.com/1682452/179742351-4bc5535b-464e-4ded-9148-29e8eb4279e6.png">

**Comment by hanneslsm** [comment]
I have created a new issue for the missing confirmation when deleting a template.

I think most of the user do not distinguish between "clear" and "delete". They will lose all their work, no matter what they click, so a similar/same implementation should be used here for both. 
Making the "Clear customisation" buttons red, as for "Delete", is definitely an option.

@jameskoster re while editing: I think a confirmation step should also be included while editing, not only for consistency but also for good UX. 
Even better would it be, if the Undo affordance would be also implemented in the Template list.

**Comment by hanneslsm** [comment]
Another random thought: If a modal is implemented, the "Export Theme" function could be linked, so that the user can download a snapshot of the theme before clearing the customisations/ deleting a template. 

Example wording, this must surely be refined…
> **Are you sure you want to clear all customisations?**
> 
> This action will remove any changes you have made and restore the original settings of the current theme. You can _copy all blocks_ or _download the complete_ theme with all templates and styles beforehand."

> **Are you sure you want to delete this template?**
> 
> Deleting this template will remove this template. This cannot be undone. You can _copy all blocks_ or _download the complete_ theme with all templates and styles beforehand."
